plugin/src/main/java/org/opensearch/ml/sdkclient/DDBOpenSearchClient.java

```diff
@@ -273,16 +304,16 @@ public CompletionStage<DeleteDataObjectResponse> deleteDataObjectAsync(DeleteDat
             .build();
         return CompletableFuture.supplyAsync(() -> AccessController.doPrivileged((PrivilegedAction<DeleteDataObjectResponse>) () -> {
             try {
-                // TODO need to return SEQ_NO here
-                // If document doesn't exist, increment and return highest seq no ever seen, but we would have to track seqNo here
-                // If document never existed, return -2 (unassigned) for seq no (probably what we have to do here)
-                // If document exists, increment and return SEQ_NO
-                dynamoDbClient.deleteItem(deleteItemRequest);
-                // TODO need to pass seqNo to simulated response
+                DeleteItemResponse deleteItemResponse = dynamoDbClient.deleteItem(deleteItemRequest);
+                Long sequenceNumber = null;
+                if (deleteItemResponse.attributes() != null && deleteItemResponse.attributes().containsKey(SEQ_NO_KEY)) {
+                    sequenceNumber = Long.parseLong(deleteItemResponse.attributes().get(SEQ_NO_KEY).n());
```

Comment:
The seqNo should be incremented on a deletion; yes, it's gone and it'll end up `-2` later, but any existing testing will expect a delete response to increment this.

Reply:
Got it, increase seq_no in response object.